eve/main.yml

```diff
@@ -290,6 +290,11 @@ models:
           [[ ${STEP_NAME:-} ]] && BUILD_STATUS_DIR+="/build_status/$STEP_NAME"
           mkdir -p "$BUILD_STATUS_DIR"
           echo -n "$FINAL_STATUS" > "$BUILD_STATUS_DIR/.final_status"
+          if [[ ${STEP_NAME:-} ]]; then
+              git checkout "%(prop:branch)s" && \
+              eve/generate_junit_result.sh \
+                  > "$BUILD_STATUS_DIR/junit_status.xml"
+          fi
```

Comment:
Oh, I see, we're only generating the report for the "Upload .final_status" step, relying on the `FINAL_STATUS` env var to know the "step" outcome (while in practice, we're using it to show the "stage" outcome).

Definitely a hack, but a good enough first step!